pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Producer.java

```diff
@@ -279,8 +280,12 @@ public void completed(Exception exception, long ledgerId, long entryId) {
                         ? ServerError.TopicTerminatedError : ServerError.PersistenceError;
 
                 producer.cnx.ctx().channel().eventLoop().execute(() -> {
-                    producer.cnx.ctx().writeAndFlush(Commands.newSendError(producer.producerId, sequenceId, serverError,
-                            exception.getMessage()));
+                    if (!(exception instanceof TopicClosedException)) {
```

Comment:
Here, broker may have sent close producer response but what is the downside here if we still send a response back? I think in old client, it didn't support close-producer command so, in that case shouldn't we send this response back?

Reply:
> Here, broker may have sent close producer response but what is the downside here if we still send a response back?

There's no need to send a bunch of `PersistenceError` back to the client in this case. Also the current reaction of the client when receiving these errors is to close connection and reconnect which will be overkill here.

> I think in old client, it didn't support close-producer command so, in that case shouldn't we send this response back?

If we saw the client didn't support the `CloseProducer`  command, we would have anyway closed the connection, so even in this case we don't need to propagate back the send-errors.